src/Build/Evaluation/Conditionals/Scanner.cs

```diff
@@ -333,17 +331,20 @@ private static bool ScanForPropertyExpressionEnd(string expression, int index, o
                         if (character == '(')
                         {
                             nestLevel++;
-                            whitespaceCheck = true;
+                            if (index + 1 < expression.Length && char.IsWhiteSpace(pchar[index + 1]) && ChangeWaves.AreFeaturesEnabled(ChangeWaves.Wave16_10))
```

Comment:
we only care about a space between the parentheses? What about before / after each one? That's the case here: https://github.com/dotnet/msbuild/issues/5863
`<ItemGroup Condition="$(Foo.StartsWith ('A'))">`

Reply:
Right, so we should ignore cases like that because we don't want them to fail.